apps/api/src/domain/models/context/edge-connection.add-audio-for-note.spec.ts

```diff
@@ -0,0 +1,139 @@
+import { AggregateType, LanguageCode } from '@coscrad/api-interfaces';
+import { bootstrapDynamicTypes } from '@coscrad/data-types';
+import assertErrorAsExpected from '../../../lib/__tests__/assertErrorAsExpected';
+import getValidAggregateInstanceForTest from '../../__tests__/utilities/getValidAggregateInstanceForTest';
+import { buildMultilingualTextFromBilingualText } from '../../common/build-multilingual-text-from-bilingual-text';
+import { buildMultilingualTextWithSingleItem } from '../../common/build-multilingual-text-with-single-item';
+import buildDummyUuid from '../__tests__/utilities/buildDummyUuid';
+import { CannnotAddAudioForNoteInGivenLanguageError } from '../context/errors';
+import { CannotOverrideAudioForLanguageError } from '../shared/multilingual-audio/errors';
+import { MultilingualAudio } from '../shared/multilingual-audio/multilingual-audio.entity';
+import { EdgeConnectionContextUnion } from './edge-connection-context-union';
+import { EdgeConnection } from './edge-connection.entity';
+import { GeneralContext } from './general-context/general-context.entity';
+import { PointContext } from './point-context/point-context.entity';
+import { TimeRangeContext } from './time-range-context/time-range-context.entity';
+
+const originalLanguageCode = LanguageCode.English;
+
+const translationLanguageCode = LanguageCode.Chilcotin;
+
+const existingTranslationOfNote = 'existing translation of note';
+
+const existingTextNote = 'existing text note';
+
+const existingBilingualNote = buildMultilingualTextFromBilingualText(
+    { text: existingTextNote, languageCode: originalLanguageCode },
+    { text: existingTranslationOfNote, languageCode: translationLanguageCode }
+);
+
+const existingMonolingualNote = buildMultilingualTextWithSingleItem(
+    existingTextNote,
+    originalLanguageCode
+);
+
+const existingEdgeConnectionWithoutAudioForNote = getValidAggregateInstanceForTest(
+    AggregateType.note
+).clone({
+    note: existingBilingualNote,
+    audioForNote: MultilingualAudio.buildEmpty(),
+});
+
+const audioItemId = buildDummyUuid(112);
+
+describe(`EdgeConnection.AddAudioForNote`, () => {
+    beforeAll(() => {
+        // TODO: use the edge connection module as a source of truth context
+        bootstrapDynamicTypes([
+            EdgeConnectionContextUnion,
+            GeneralContext,
+            TimeRangeContext,
+            PointContext,
+            EdgeConnection,
+        ]);
+    });
+    describe(`when the update is valid`, () => {
```

Comment:
Do these cases pass? I didn't see the actual implementation for this in the entity class in the update method.

Reply:
all cases pass except my last one